### PR TITLE
Scoreboard UI

### PR DESCRIPTION
**Pull request by walker407**
Adds the ability to display a scoreboard menu when the player presses the "score" key (Tab).  Currently, the scoreboard shows the following:

Server name
List of connected client names and ping

### CODE REVIEW

**Review by Remscar — APPROVED** [review]

File: code/ui/scoreboard/Scoreboard.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    public void Close()
+    {
+        IsOpen = false;
+    }
+    protected override int BuildHash()
+    {
+        var queueHash = 11;
+
+        return HashCode.Combine(IsOpen, queueHash, Clients, Clients.Count);
```
Comment (Remscar) [inline code/ui/scoreboard/Scoreboard.cs:34]: Just throwing `Clients` in here won't actually work the way you intend it to.

You'd have to look through each client and compound a hash with the properties you actually care about.

File: code/ui/scoreboard/Scoreboard.razor [hunk 2]
```diff
@@ -0,0 +1,145 @@
+﻿@using System.Collections.Generic;
+
+@namespace Cinema.UI
+
+<style>
```
Comment (Remscar) [inline code/ui/scoreboard/Scoreboard.razor:5]: I recommend that you split the css to a .scss file

Comment (walker407) [reply code/ui/scoreboard/Scoreboard.razor]: Sounds good, I moved the styling out of the razor file to a .scss file in the newest commit.

File: code/ui/scoreboard/Scoreboard.cs (hunk 1 above)
Comment (walker407) [reply code/ui/scoreboard/Scoreboard.cs:34]: I'm thinking about updating the hash function with the code snippet below. Does it look alright? 

```
protected override int BuildHash()
{
    var clientHash = 11;

    foreach(var client in Clients)
    {
        clientHash = clientHash * 31 + client.GetHashCode();
    }

    return HashCode.Combine(IsOpen, Clients.Count, clientHash);
}

**Pull request merged**